ext/openssl/openssl.c

```diff
@@ -4202,14 +4206,15 @@ PHP_FUNCTION(openssl_pkey_export_to_file)
 	}
 	RETVAL_FALSE;
 
-	PHP_OPENSSL_CHECK_SIZE_T_TO_INT(passphrase_len, passphrase);
-	key = php_openssl_evp_from_zval(zpkey, 0, passphrase, passphrase_len, 0, &key_resource);
+	PHP_OPENSSL_CHECK_SIZE_T_TO_INT(passphrase_len, passphrase, 3);
 
+	key = php_openssl_evp_from_zval(zpkey, 0, passphrase, passphrase_len, 0, &key_resource);
 	if (key == NULL) {
 		if (!EG(exception)) {
-			php_error_docref(NULL, E_WARNING, "Cannot get key from parameter 1");
+			// TypeError?
+			zend_argument_value_error(1, "cannot retrieve key");
```

Comment:
Something like:
```suggestion
			zend_argument_value_error(1, "doesn't have a key");
```
I'm not exactly sure how to phrase this though

Reply:
That doesn't seem to be correct looking at the implementation of  ``php_openssl_evp_from_zval()`` which looks it can fail for weird reasons.